R/plotPropselRange.R

```diff
@@ -0,0 +1,144 @@
+#' @importFrom graphics lines
+#' @importFrom lavaan cfa HolzingerSwineford1939
+NULL
+
+#' Plot classification accuracy indices at different proportions of selection
+#' 
+#' \code{plotPropselRange} plots classification accuracy indices at different
+#' proportions of selection under partial and strict invariance conditions for 
+#' a given CFA fit.
+#' 
+#' @param cfa_fit CFA model output from lavaan.
+#' @param from The lowest proportion of selection to consider
+#' @param to The largest proportion of selection to consider
+#' @param by The increment of the sequence of proportions.
+#' @param pmix List of length `g` containing the mixing proportions of each
+#'     group (where `g` is the number of groups). If `NULL`, defaults to `1/g` 
+#'     for each group (i.e., the populations have equal size).
+#' @param labels A character vector with `g` elements to label the reference
+#'     and focal groups on the plot, where `g` is the number of groups. If not
+#'     provided, groups are labeled automatically as 'Reference' (for the first
+#'     group) and 'Focal_1' through 'Focal_(g-1)', where `g` is the number of
+#'     groups.
+#' @return Eight plots illustrating how proportion selected (PS), success ratio 
+#'     (SR), sensitivity (SE), and specificity (SP) change across different 
+#'     proportions of selection under partial and strict invariance conditions.
+#' @examples
+#' \dontrun{
+#' HS <- HolzingerSwineford1939
+#' HS$sex <- as.factor(HS$sex)
+#' HS.model <- ' visual  =~ x1 + x2 + x3
+#'               textual =~ x4 + x5 + x6
+#'               speed   =~ x7 + x8 + x9 '
+#' 
+#' fit <- cfa(HS.model, data = HS, group = "sex")
+#' 
+#' plotPropselRange(fit, pmix = table(HS$sex)/sum(table(HS$sex)))
+#' }
+#' @export
+plotPropselRange <- function(cfa_fit,
+                             from = 0.01,
+                             to = 0.25,
+                             by = 0.01,
+                             pmix = NULL,
+                             labels = NULL) {
+  est <- format_cfa_partinv(cfa_fit)
```

Comment:
This now throws an error saying "argument "comp" is missing, with no default"

Reply:
thanks for catching, I just added comp = "est" to the function call